src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.cs

```diff
@@ -0,0 +1,311 @@
+
+        // Receives a byte array that represents an ASCII string containing a number in octal base.
+        // Converts the array to an octal base number, then transforms it to ten base and returns it.
+        internal static int GetTenBaseNumberFromOctalAsciiChars(Span<byte> buffer)
+        {
+            string str = GetTrimmedAsciiString(buffer);
+            return string.IsNullOrEmpty(str) ? 0 : Convert.ToInt32(str, fromBase: 8);
+        }
+
+        // Returns the string contained in the specified buffer of bytes,
+        // in the specified encoding, removing the trailing null or space chars.
+        private static string GetTrimmedString(ReadOnlySpan<byte> buffer, Encoding encoding)
+        {
+            int trimmedLength = buffer.Length;
+            while (trimmedLength > 0 && IsByteNullOrSpace(buffer[trimmedLength - 1]))
+            {
+                trimmedLength--;
+            }
+
+            return trimmedLength == 0 ? string.Empty : encoding.GetString(buffer.Slice(0, trimmedLength));
+
+            static bool IsByteNullOrSpace(byte c) => c is 0 or 32;
+        }
+
+        // Returns the ASCII string contained in the specified buffer of bytes,
+        // removing the trailing null or space chars.
+        internal static string GetTrimmedAsciiString(ReadOnlySpan<byte> buffer) => GetTrimmedString(buffer, Encoding.ASCII);
+
+        // Returns the UTF8 string contained in the specified buffer of bytes,
+        // removing the trailing null or space chars.
+        internal static string GetTrimmedUtf8String(ReadOnlySpan<byte> buffer) => GetTrimmedString(buffer, Encoding.UTF8);
+
+        // Reads the specified number of bytes and stores it in the byte buffer passed by reference.
+        // Throws if end of stream is reached.
+        internal static void ReadOrThrow(Stream archiveStream, Span<byte> buffer)
+        {
+            int totalRead = 0;
+            while (totalRead < buffer.Length)
+            {
+                int bytesRead = archiveStream.Read(buffer.Slice(totalRead));
+                if (bytesRead == 0)
+                {
+                    throw new EndOfStreamException();
+                }
+                totalRead += bytesRead;
+            }
+        }
+
+        // Returns true if it successfully converts the specified string to a DateTimeOffset, false otherwise.
+        internal static bool TryConvertToDateTimeOffset(string value, out DateTimeOffset timestamp)
+        {
+            timestamp = default;
+            if (!string.IsNullOrEmpty(value))
+            {
+                if (!double.TryParse(value, NumberStyles.Any, CultureInfo.InvariantCulture, out double doubleTime))
+                {
+                    return false;
+                }
+
+                try
+                {
+                    timestamp = GetDateTimeFromSecondsSinceEpoch(doubleTime);
+                }
+                catch // TODO: Remove this temporary exception to verify unexpected culture value
```

Comment:
?

Reply:
I had to add this to be able to investigate the culture problem happening only in the CI. I wanted to learn what was being generated. I won't merge it.